d2-schemas/src/main/pegasus/com/linkedin/d2/D2TransportClientProperties.pdsc

```diff
@@ -16,6 +16,12 @@
       "doc": "Timeout in ms for this transport client. Defaults to 10s.",
       "optional": true
     },
+    {
+      "name": "streamingTimeout",
+      "type": "long",
+      "doc": "Streaming Timeout in ms for this transport client. Disabled by default (-1)",
```

Comment:
Why -1? 0 should disable it as well, right?

Reply:
The streaming implementation expect -1, 0 will make the request immediately timeout.

Reply:
We probably do not have any use case to immediately timeout a request. Prefer to have 0 to disable it as other configs. But that probably belong to another changes.